src/en/guide/upgrading/upgrading_4_0_x.adoc

```diff
@@ -0,0 +1,80 @@
+### Bump up Grails Version
+
+You will need to upgrade your Grails version defined in `gradle.properties` as:
+
+[source,groovy,subs="attributes"]
+.gradle.properties
+----
+...
+grailsVersion=4.1.0.M1
+...
+----
+
+### Groovy 3.0.3
+
+Grails 4.1 provide support for Groovy 3. We would recommend you to please check the https://groovy-lang.org/releasenotes/groovy-3.0.html[Release notes for Groovy 3] to update your application in case you are using a specific feature which might not work in Groovy 3.
+
+Define groovyVersion in `gradle.properties` to force the application to use Groovy 3.
+
+Grails 4.1 app's `gradle.properties`
+
+[source, properties]
+.gradle.properties
+----
+...
+groovyVersion=3.0.3
+...
+----
+
+### Bump up GORM Version
+
+If you were using http://gorm.grails.org[GORM], you will need to update the version defined in `gradle.properties` as:
+
+[source, properties]
+.gradle.properties
+----
+...
+gormVersion=7.1.0.M1
+...
+----
+
+### Bump up Asset Pipeline plugin version
+
+The previous version of asset-pipeline is not supported with Grails 4.1 as it is compiled with a version of Groovy which is binary incompatible with Groovy 3. So, please update the plugin version to 3.1.0.
+
+### Disabled StringCharArrayAccessor by default
+
+The previous version of Grails use the `StringCharArrayAccessor` which is enabled by default and provides optimized access to `java.lang.String` internals. In Grails 4.1 it is disabled by default but you can enable it by setting a system property with name `stringchararrayaccessor.disabled` and value `false`.
+
+NOTE: Enabling StringCharArrayAccessor would show IllegalReflectiveAccess warnings as it uses reflection to do the optimizations.
+
+### Changes in profile.yml and feature.yml files in Grails Profiles
+
+The format of how dependencies are defined in features and profiles has been changed. See the section on link:{guidePath}/profiles.html[Application Profiles] for more information.
+
+### Deprecation of dot navigation of Grails configuration
+
+In order to reduce complexity, improve performance, and increase maintainability, accessing configuration through dot notation (config.a.b.c) has been deprecated. This functionality will be removed in a future release.
```

Comment:
What is the reason for removing this functionality?  This has been in Grails for a long time and removing to reduce complexity at the expense of requiring all apps to change seems counter-intuitive.

Reply:
@bobbywarner Applications should have changed a long time ago when the `getProperty` methods were introduced. This is deprecated, so there will be plenty of time for folks to migrate.